www/docs/client/links/httpSubscriptionLink.md

```diff
@@ -140,56 +140,59 @@ import {
   httpBatchLink,
   splitLink,
   unstable_httpSubscriptionLink,
+  retryLink,
 } from '@trpc/client';
 import {
   EventSourcePolyfill,
   EventSourcePolyfillInit,
 } from 'event-source-polyfill';
 import type { AppRouter } from '../server/index.js';
 
-// polyfill EventSource
-globalThis.EventSource = EventSourcePolyfill;
 
 // Initialize the tRPC client
 const trpc = createTRPCClient<AppRouter>({
   links: [
     splitLink({
       condition: (op) => op.type === 'subscription',
-      true: unstable_httpSubscriptionLink({
-        url: async () => {
-          // calculate the latest URL if needed...
-          return getAuthenticatedUri();
-        },
-        eventSourceOptions: async () => {
-          // ...or maybe renew an access token
-          const token = await auth.getOrRenewToken();
-
-          return {
-            headers: {
-              authorization: 'Bearer ' + token,
-            },
-          } as EventSourcePolyfillInit;
-        },
-
-        // In this example we handle an authentication failure
-        experimental_shouldRecreateOnError(opts) {
-          let willRestart = false;
-          if (opts.type === 'event') {
-            const ev = opts.event;
-            willRestart =
-              'status' in ev &&
-              typeof ev.status === 'number' &&
-              [401, 403].includes(ev.status);
-          }
-          if (willRestart) {
-            console.log('Restarting EventSource due to 401/403 error');
-          }
-          return willRestart;
-        },
-      }),
       false: httpBatchLink({
         url: 'http://localhost:3000',
       }),
+      true: [
+        retryLink({
+          retry: (opts) => {
+            let willRestart = false;
+
+            if (
+              opts.error.data?.code === 'UNAUTHORIZED' ||
+              opts.error.data?.code === 'FORBIDDEN'
+            ) {
+              willRestart = true;
+              console.log(
+                'Restarting EventSource due to 401/403 error',
+              );
+            }
+            return willRestart;
```

Comment:
```suggestion
            if (
              opts.error.data?.code === 'UNAUTHORIZED' ||
              opts.error.data?.code === 'FORBIDDEN'
            ) {
              console.log(
                'Restarting EventSource due to 401/403 error',
              );
              return true
            }
```

Bit of a nit, but the code seems unnecessarily convoluted and may leave folk wondering why it's considered a best practice to write it that way.

Not sure if an undefined return will pass typecheck, so can also add `return false` on the end if not

Reply:
I mainly followed your suit here! :D